impeller/entity/entity_pass.cc

```diff
@@ -776,15 +846,19 @@ bool EntityPass::RenderElement(Entity& element_entity,
   }
 
   if (result.just_created) {
-    // Restore any clips that were recorded before the backdrop filter was
-    // applied.
-    auto& replay_entities = clip_coverage_stack.GetReplayEntities();
-    for (const auto& replay : replay_entities) {
-      SetClipScissor(clip_coverage_stack.CurrentClipCoverage(), *result.pass,
-                     global_pass_position);
-      if (!replay.entity.Render(renderer, *result.pass)) {
-        VALIDATION_LOG << "Failed to render entity for clip restore.";
-      }
+    clip_coverage_stack.ActivateClipReplay();
+  }
+
+  // If there are any pending clips to replay, render any that may affect
+  // the entity we're about to render.
+  while (const EntityPassClipStack::ReplayResult* next_replay_clip =
+             clip_coverage_stack.GetNextReplayResult(element_entity)) {
+    auto& replay_entity = next_replay_clip->entity;
+    SetClipScissor(next_replay_clip->clip_coverage, *result.pass,
```

Comment:
This is the only line that changed from the previous attempt (other than the new golden that I added).

`clip_coverage_stack.CurrentClipCoverage()` changed to `next_replay_clip->clip_coverage`.